azurelinuxagent/common/protocol/wire.py

```diff
@@ -1121,33 +1138,32 @@ def report_event(self, event_list):
                                          .format(str(details_of_event), len(event_str), MAX_EVENT_BUFFER_SIZE))
                     continue
                 if len(buf[event.providerId] + event_str) >= MAX_EVENT_BUFFER_SIZE:
-                    self.send_encoded_event(event.providerId, buf[event.providerId])
+                    logger.verbose("No of events this request = {0}".format(events_per_request[event.providerId]))
+                    uni_err_count, err_count = _send_event(event.providerId)
+                    unicode_error_count += uni_err_count
+                    event_report_error_count += err_count
                     buf[event.providerId] = b''
-                    logger.verbose("No of events this request = {0}".format(events_per_request))
-                    events_per_request = 0
+                    events_per_request[event.providerId] = 0
```

Comment:
Would it be worth it to extract this functionality into a function? Though it isn't repeated in code, I still feel like giving it a name would help with readability.

Reply:
I totally get what you mean, it does seem pretty convoluted. Let me see what I can do :)

Reply:
Done